keyv/Ceph.cpp

```diff
@@ -176,118 +78,272 @@ class Ceph : public Plugin
         librados::bufferlist bl;
     };
 
-    typedef stde::hash_map<std::string, AsyncRead> ReadMap;
+    typedef std::unordered_map<std::string, AsyncRead> ReadMap;
     typedef std::deque<librados::AioCompletion*> Writes;
 
     mutable ReadMap _reads;
     Writes _writes;
 
-    bool _fetch(const std::string& key, const size_t sizeHint) const
-    {
-        if (_reads.size() >= _maxPendingOps)
-            return true;
+    bool _fetch(const std::string& key, const size_t sizeHint) const;
+
+    bool _flush(const size_t maxPending);
+
+    librados::bufferlist _get(const std::string& key) const;
+};
 
-        AsyncRead& asyncRead = _reads[key];
-        if (asyncRead.op)
-            return true; // fetch for key already pending
+inline Ceph::Ceph(const servus::URI& uri)
+    : _maxPendingOps(0)
```

Comment:
I will put 256 which corresponds to the best performance that I could obtain. But still arbitrary value.